### PR TITLE
add docs and update README.md

### PR DESCRIPTION
**Pull request by ns408**
- also removed extra whitespace characters.

### CODE REVIEW

File: docs/howto_contribute.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Contributing
+
+The page describes how we work with this repository.
+
+## General principles
+
+- Please don't push changes to the master directly. Allow your changes to be peer reviewed.
+- Please use `dev` branch for things that may break existing features.
+- Please use `stable` branch for production deployments.
```
Comment (patchkez) [inline docs/howto_contribute.md]: Can we use `master` branch for production/stable code? I think no one has merge rights into this repo except me, but also I will make PRs so other can review changes.

File: docs/vision.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Vision
+
+This document is for describing the vision. We may eventually end up separating this with a higher level architectural diagram and a separate doc for implementation details.
+
+## Why do we need containerisation
+
+- On komodoplatform, we run a whole range of chains which get compiled, configured and deployed differently. Each of these may have different dependencies and it gets harder to test things in isoltation when we deploy all of these on a single server. We  would like to handle this problem through containerisation where self container docker images can be used to launch containers which work exactly the same irrespective of underlying operating system (and it's version).
+- Compilation takes a while and occasionally we see 3rd party packages being missed. We would like to keep dockerhub images which have compiled version that can just be used without any fuss.
+- Local testing on your laptops gets easier because you can just work with a container, make changes and then get it deployed onto your production environment.
```
Comment (patchkez) [inline docs/vision.md:9]: I would mention here also scalability. With growing number of coins, there will be need to spread load between multiple nodes. I think this article is still valid https://dragonriders.team/tech/2018/03/20/proposed-notary-node-architecture.html

**Review by patchkez — COMMENTED** [review]
Looks good, I am fine with merging once we decide what branch to use for PROD.

File: docs/howto_contribute.md (hunk 1 above)
Comment (ns408) [reply docs/howto_contribute.md]: @patchkez Sure, let's stick with `master` then. I'm gonna update the PR then.

As for you creating PRs, sounds good.

File: docs/vision.md (hunk 2 above)
Comment (ns408) [reply docs/vision.md:9]: @patchkez Good point. Updating the doco to include it.

**Pull request merged**